crypt/cipher.go

```diff
@@ -284,11 +293,182 @@ func (c *cipher) decryptSegment(ciphertext string) (string, error) {
 	return string(plaintext), err
 }
 
+// Simple obfuscation routines
+
+func (c *cipher) obfuscateSegment(plaintext string) string {
+	if plaintext == "" {
+		return ""
+	}
+
+	// If the string isn't valid UTF8 then don't rotate; just
+	// prepend a 0.
+	if !utf8.ValidString(plaintext) {
+		return "0." + plaintext
+	}
+
+	// Calculate a simple rotation based on the filename and
+	// the nameKey
+	var dir int
+	for _, runeValue := range plaintext {
+		dir += int(runeValue)
+	}
+	dir = dir % 256
+
+	// We'll use this number to store in the result filename...
+	result := strconv.Itoa(dir) + "."
+
+	// but we'll augment it with the nameKey for real calculation
+	for i := 0; i < len(c.nameKey); i++ {
+		dir += int(c.nameKey[i])
+	}
+	dir = dir % 127
+
+	// Now for each character, depending on the range it is in
+	// we will actually rotate a different amount
+	var newChar string
+
+	for _, runeValue := range plaintext {
+		switch {
+		case string(runeValue) == obfuscQuote:
+			// Quote the Quote character
+			newChar = obfuscQuote + obfuscQuote
+
+		case (runeValue >= 0x30 && runeValue <= 0x39):
+			// Number
+			thisdir := (dir % 9) + 1
+			newRune := 0x30 + (int(runeValue)-0x30+thisdir)%10
+			newChar = string(newRune)
+
+		case (runeValue >= 0x41 && runeValue <= 0x5A) ||
+			(runeValue >= 0x61 && runeValue <= 0x7A):
+			// ASCII letter.  Try to avoid trivial A->a mappings
+			thisdir := dir%(obfuscASCIILen/2-1) + 1
+			pos := strings.Index(obfuscASCII, string(runeValue))
```

Comment:
I want to be able to potentially map upper -> lower (eg X->f) but I think I've learned enough Go now to be able to write that as a formula instead of an `Index`.